### PR TITLE
Set the username / password of the Geoserver's admin user via env variables - fix

### PR DESCRIPTION
**Pull request by athanasp**
#### Reference issue

https://github.com/kartoza/docker-geoserver/issues/326

#### Description

Set the username / password of the Geoserver's admin user from the enviroment variables `GEOSERVER_ADMIN_USER` and `GEOSERVER_ADMIN_PASSWORD` (or `GEOSERVER_ADMIN_PASSWORD_FILE`).

Docker image tag: `kartoza/geoserver:2.20.0`
Deployment environment: `kubernetes`

### CODE REVIEW

File: scripts/functions.sh [hunk 1]
```diff
@@ -260,7 +260,7 @@ function geoserver_logging() {
 function file_env {
 	local var="$1"
 	local fileVar="${var}_FILE"
-	local def="${2:-}"
+	local def="${1:-}"
```
Comment (NyakudyaA) [inline scripts/functions.sh:263]: Does this fix it for all env vars that can be called with `_FILE` ?

**Review by NyakudyaA — COMMENTED** [review]
@athanasp   Can you address the comments

File: scripts/functions.sh (hunk 1 above)
Comment (athanasp) [reply scripts/functions.sh:263]: Given the fact that this function takes just one argument in the [env-data.sh](https://github.com/kartoza/docker-geoserver/blob/master/scripts/env-data.sh) bash script, I expect to fix any env variable with **_FILE** suffix.

File: scripts/entrypoint.sh [hunk 2]
```diff
@@ -8,6 +8,7 @@ source /scripts/functions.sh
 source /scripts/env-data.sh
 
 /bin/bash /scripts/start.sh
+/bin/bash /scripts/update_passwords.sh
```
[the remark this answers is not in the document]
Comment (athanasp) [reply scripts/entrypoint.sh:11]: Initially, i deployed the geoserver without the `/bin/bash /scripts/update_passwords.sh` in the `scripts/entrypoint.sh` but it didn't work. By adding explicity the line 11 in the `scripts/entrypoint.sh` the password was updated and then, i could log in the geoserver admin UI using the password from env variable.

Do you want to check it again?

[the remark this answers is not in the document]
Comment (NyakudyaA) [reply scripts/entrypoint.sh:11]: You can proceed to change it that way by adding `/bin/bash` See https://github.com/kartoza/docker-geoserver/issues/326#issuecomment-976182094

**Comment by NyakudyaA** [comment]
@athanasp can you try to build the image from master and tell me if it works changing the password. If so we can adjust this PR to cater for one of the issue being addressed here

**Comment by NyakudyaA** [comment]
@athanasp can you confirm if master fixes the password issue. I suggest you pull from master and test this so that we can merge this if it's still needed

**Pull request closed without merging**